File: app/src/main/java/torille/fi/lurkforreddit/utils/Store.java
package torille.fi.lurkforreddit.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by eva on 23.4.2017.
 */
@Singleton
final public class Store {

    private SharedPreferences mSharedPreferences;

    private static final String TOKEN = "token";
    private static final String REFRESH_TOKEN = "refreshtoken";
    private static final String LOGGED_IN = "logged_in";

    @Inject
    public Store(@NonNull Context context) {
        context = context.getApplicationContext();
        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public String getToken() {
        String token = mSharedPreferences.getString(TOKEN, null);
        if (token == null) {
            return "";
        }
        return token;
    }

    public void setToken(String token) {
        mSharedPreferences.edit().putString(TOKEN, token).apply();
    }

    public void setRefreshToken(String token) {
        mSharedPreferences.edit().putString(REFRESH_TOKEN, token).apply();
    }

    public String getRefreshToken() {
        return mSharedPreferences.getString(REFRESH_TOKEN, null);
    }

    public Boolean isLoggedIn() {
        return mSharedPreferences.getBoolean(LOGGED_IN, false);
    }

    public void loggedIn(Boolean loggedIn) {
        mSharedPreferences.edit().putBoolean(LOGGED_IN, loggedIn).apply();
    }
}

File: app/src/main/java/torille/fi/lurkforreddit/utils/Store.kt
package torille.fi.lurkforreddit.utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Utility class to get values from SharedPreferences
 */
@Singleton
class Store @Inject
constructor(context: Context) {

    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context.applicationContext)

    var token: String
        get() = sharedPreferences.getString(TOKEN, "")
        set(token) = sharedPreferences.edit().putString(TOKEN, token).apply()

    var refreshToken: String
        get() = sharedPreferences.getString(REFRESH_TOKEN, "")
        set(token) = sharedPreferences.edit().putString(REFRESH_TOKEN, token).apply()

    val isLoggedIn: Boolean
        get() = sharedPreferences.getBoolean(LOGGED_IN, false)

    fun loggedIn(loggedIn: Boolean?) {
        sharedPreferences.edit().putBoolean(LOGGED_IN, loggedIn!!).apply()
    }

    companion object {

        private val TOKEN = "token"
        private val REFRESH_TOKEN = "refreshtoken"
        private val LOGGED_IN = "logged_in"
    }
}
